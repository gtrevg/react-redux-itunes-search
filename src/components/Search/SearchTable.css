.SearchTable {
  padding: 5px;
  margin: 10px;
  border-radius: 8px;
}

.SearchTableTitle {
    width: 25%;
    min-width: 160px;
}

.SearchTableArtist {
    width: 20%;
    min-width: 120px;
}

.SearchTableAlbum {
    width: 25%;
    min-width: 160px;
}

.SearchTableRelease {
    width: 5%;
    min-width: 120px;
}

.SearchTableCover {
    width: 5%;
    min-width: 80px;
}

.SearchTableLenght {
    width: 5%;
    min-width: 80px;
}

.SearchTableGenre {
    width: 10%;
    min-width: 120px;
}

.SearchTablePrice {
    width: 5%;
    min-width: 120px;
}

@media (max-width: 1240px) {
  .SearchTablePrice {
    display: none;
  }
}

@media (max-width: 1120px) {
  .SearchTableGenre {
    display: none;
  }
}

@media (max-width: 1040px) {
  .SearchTableLenght {
    display: none;
  }
}

@media (max-width: 960px) {
  .SearchTableCover {
    display: none;
  }
}

@media (max-width: 840px) {
  .SearchTableRelease {
    display: none;
  }
}
