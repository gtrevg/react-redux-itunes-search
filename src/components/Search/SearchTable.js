import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Table } from 'antd';
import moment from 'moment';

import './SearchTable.css';

const getDate = text => moment(text).format('DD/MM/YYYY');

const getDuration = text => {
	const minutes = moment.duration(text).minutes() + '';
	const seconds = moment.duration(text).seconds() + '';
	return minutes.padStart(2, '0') + ':' + seconds.padStart(2, '0');
};

const columns = [
	{
		title: 'Title',
		dataIndex: 'title',
		key: 'title',
		className: 'SearchTableTitle',
	},
	{
		title: 'Artist',
		dataIndex: 'artist',
		key: 'artist',
		className: 'SearchTableArtist',
	},
	{
		title: 'Album title',
		dataIndex: 'album',
		key: 'album',
		className: 'SearchTableAlbum',
	},
	{
		title: 'Release date',
		dataIndex: 'release',
		key: 'release',
		className: 'SearchTableRelease',
		render: getDate,
	},
	{
		title: 'Cover',
		dataIndex: 'cover',
		key: 'cover',
		className: 'SearchTableCover',
		render: (text, record) => (
			<img alt={record.title} src={text} style={{ width: 60 }} />
		),
	},
	{
		title: 'Lenght',
		dataIndex: 'lenght',
		key: 'lenght',
		className: 'SearchTableLenght',
		render: getDuration,
	},
	{
		title: 'Genre',
		dataIndex: 'genre',
		key: 'genre',
		className: 'SearchTableGenre',
	},
	{
		title: 'Price',
		dataIndex: 'price',
		key: 'price',
		className: 'SearchTablePrice',
	},
];

class SearchTable extends PureComponent {
	static contextTypes = {
		router: PropTypes.object,
	};

	goToDetail(id) {
		this.context.router.history.push(`/detail?id=${id}`);
	}

	render() {
		const { isLoading, dataSource } = this.props;
		return (
			<div className="SearchTable">
				<Table
					loading={isLoading}
					dataSource={dataSource}
					columns={columns}
					pagination={{ pageSize: 5 }}
					onRow={record => {
						return {
							onClick: () => this.goToDetail(record.key),
						};
					}}
				/>
			</div>
		);
	}
}

const mapStateToProps = state => {
	return {
		...state.search,
	};
};

export default connect(mapStateToProps)(SearchTable);
