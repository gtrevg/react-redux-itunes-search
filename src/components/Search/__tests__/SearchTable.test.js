import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from "react-router-dom";

import SearchTable from '../index';
import store from '../../../config/mockStore';

describe( 'SearchTable component', () => {

  const initialEntries = ['/'];
  const component =
    <Provider store={store}>
      <MemoryRouter initialEntries={initialEntries}>
        <SearchTable />
      </MemoryRouter>
    </Provider>;

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
