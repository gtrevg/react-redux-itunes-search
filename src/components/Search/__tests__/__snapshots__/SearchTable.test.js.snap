// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTable component Snapshot matchs 1`] = `
<div
  className="SearchTable"
>
  <mocked
    name="Antd Table"
    props={
      Object {
        "columns": Array [
          Object {
            "className": "SearchTableTitle",
            "dataIndex": "title",
            "key": "title",
            "sorter": [Function],
            "title": "Title",
          },
          Object {
            "className": "SearchTableArtist",
            "dataIndex": "artist",
            "key": "artist",
            "sorter": [Function],
            "title": "Artist",
          },
          Object {
            "className": "SearchTableAlbum",
            "dataIndex": "album",
            "key": "album",
            "sorter": [Function],
            "title": "Album title",
          },
          Object {
            "className": "SearchTableRelease",
            "dataIndex": "release",
            "key": "release",
            "render": [Function],
            "sorter": [Function],
            "title": "Release date",
          },
          Object {
            "className": "SearchTableCover",
            "dataIndex": "bigCover",
            "key": "bigCover",
            "render": [Function],
            "title": "Cover",
          },
          Object {
            "className": "SearchTableLenght",
            "dataIndex": "lenght",
            "key": "lenght",
            "render": [Function],
            "sorter": [Function],
            "title": "Lenght",
          },
          Object {
            "className": "SearchTableGenre",
            "dataIndex": "genre",
            "key": "genre",
            "sorter": [Function],
            "title": "Genre",
          },
          Object {
            "className": "SearchTablePrice",
            "dataIndex": "price",
            "key": "price",
            "sorter": [Function],
            "title": "Price",
          },
        ],
        "dataSource": Array [
          Object {
            "album": "Bad",
            "artist": "Michael Jackson",
            "cover": "http://is2.mzstatic.com/image/thumb/Music/v4/05/c4/1c/05c41ca0-b614-a91c-e63e-2fd9a59d19e8/source/60x60bb.jpg",
            "genre": "Pop",
            "key": 559334764,
            "lenght": 257766,
            "price": "1.29 USD",
            "release": "1987-08-31T07:00:00Z",
            "title": "Smooth Criminal",
          },
          Object {
            "album": "Thriller",
            "artist": "Michael Jackson",
            "cover": "http://is4.mzstatic.com/image/thumb/Music/v4/ec/2b/17/ec2b170d-2ecb-50af-02eb-39f0decc2847/source/60x60bb.jpg",
            "genre": "Pop",
            "key": 269573364,
            "lenght": 294188,
            "price": "1.29 USD",
            "release": "1982-11-30T08:00:00Z",
            "title": "Billie Jean",
          },
          Object {
            "album": "The Essential Michael Jackson",
            "artist": "Michael Jackson",
            "cover": "http://is2.mzstatic.com/image/thumb/Music127/v4/8a/65/be/8a65bef2-f23d-e43d-9124-f5e4293513f7/source/60x60bb.jpg",
            "genre": "Pop",
            "key": 159294478,
            "lenght": 320905,
            "price": "1.29 USD",
            "release": "1987-08-31T07:00:00Z",
            "title": "Man In the Mirror",
          },
        ],
        "loading": false,
        "onRow": [Function],
        "pagination": Object {
          "pageSize": 5,
        },
      }
    }
  />
</div>
`;
